packages/lodestar/src/network/network.ts

```diff
@@ -56,7 +58,7 @@ export class Libp2pNetwork extends (EventEmitter as { new(): NetworkEventEmitter
     // `libp2p` can be a promise as well as a libp2p object
     this.inited = new Promise((resolve) => {
```

Comment:
I don't see in which case libp2p is a promise, probably this happens in old version of libp2p only?

Reply:
Yeah, we used to maybe initialize a libp2p object in BeaconNode constructor. Since constructor is synchronous function, we couldn't await there, but would await here. Now, we properly create a libp2p object before creating a new BeaconNode, so we probably shouldn't support this anymore.